src/middleware/withAPIPageLogger.ts

```diff
@@ -0,0 +1,56 @@
+import { createLogger } from "@fewlines/fwl-logging";
+import { NextApiRequest, NextApiResponse } from "next";
+import { Handler } from "src/pages/api/identity";
+
+type WithAPIPageLogger = (
+  request: NextApiRequest,
+  response: NextApiResponse,
+) => Promise<JSON | void>;
```

Comment:
Not sure if this should be type as `Promise<JSON | Error>` and I throw an Error instead of console logging it.

Reply:
Let me see if I understand correctly, this type is the return type of `WithAPIPageLogger` so the function that will be in the end passed to next for the API page? 

Shouldn't it be always `void` in that case since we don't return via the function but via the `response.json`?

Reply:
It is, but I remember you saying that returning `response.json` should be typed as `JSON` and not `void` (both are working).